core/src/stored/butil.cc

```diff
@@ -170,8 +170,18 @@ static bool setup_to_access_device(DeviceControlRecord* dcr,
   }
 
   if ((device_resource = find_device_res(dev_name, readonly)) == NULL) {
-    Jmsg2(jcr, M_FATAL, 0, T_("Cannot find device \"%s\" in config file %s.\n"),
-          dev_name, configfile);
+    std::string available_devices;
+    for (BareosResource* resource = nullptr;
+         (resource = my_config->GetNextRes(R_DEVICE, resource));) {
+      available_devices += '\"';
+      available_devices += resource->resource_name_;
+      available_devices += '\"';
+      available_devices += '\n';
```

Comment:
As we also allow to specify the `archive device` instead of the name of the resource, we should print it as well.
```suggestion
      available_devices += ' \"';
      available_devices += resource->resource_name_;
      available_devices += '\" (';
      available_devices += resource->archive_device_string
      available_devices += ')\n';
```
I would also like some indentation to make the list easier to see.